main.tf

```
@@ -118,6 +118,18 @@ resource "github_repository" "repository" {
     }
   }
 
+  dynamic "pages" {
+    for_each = var.pages != null ? [true] : []
+
+    content {
+      source {
+        branch = try(var.pages.branch, null)
```

Could this default to `gh-pages`?

---

you mean that we should rename the variable to `gh-pages` vs `pages`?

---

Hmm, no I don't think so. It should match the underlying resources API. What we can do so, is to add it to the defaults. So indeed, you may configure `gh-pages` as the default branch for GitHub Pages across all your repositories. See https://github.com/mineiros-io/terraform-github-repository#defaults-object-attributes for details. WDYT?

---

although, it would mean that gh pages will be enabled for all repos automatically - I don't think that's a good idea anyways.

---

Sorry, I meant the default pages branch. It defaults to `gh-pages` in GitHub UI.

---

not sure it makes sense to define a branch as it might fail if it does not exist before... we should test this and maybe set the desired default in a later release.

but for sure branch ist not optional but required so please change to

```suggestion
        branch = var.pages.branch
```

---

resolved in 1f98adf

---

will think about the default branch - if we set the path and branch with default values, we need another trigger to keep gh pages deactivated since it will otherwise always be activated using the default settings.